### PR TITLE
Add python/django invocations

### PR DESCRIPTION
**Pull request by TheSuperiorStanislav**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -186,3 +186,110 @@ Add hosts to `/etc/hosts`.
 
 Settings:
 * `github_action_hosts` image tag of builder (Default: see `docker-main-containers`)
+
+### python
+
+#### run_python
+
+Run python command depending on `PYTHON_ENV` variable(`docker` or `local`).
```
Comment (ron8mcr) [inline README.md:194]: Let's add separate doc about different python envs, motivations, usage examples

**Review by ron8mcr — CHANGES_REQUESTED** [review]

File: README.md [hunk 2]
```diff
@@ -186,3 +186,110 @@ Add hosts to `/etc/hosts`.
...
+#### check_new_migrations
+
+Check if there is new migrations or not.
+
+#### migrate
+
+Run `migrate` command.
+
+Settings:
+* `django_migrate_command` migrate command (Default: `migrate`)
```
[the remark this answers is not in the document]
Comment (TheSuperiorStanislav) [reply README.md:225]: The one that uses `django-tenats`

File: README.md [hunk 3]
```diff
@@ -186,3 +186,123 @@ Add hosts to `/etc/hosts`.
 
 Settings:
 * `github_action_hosts` image tag of builder (Default: see `docker-main-containers`)
+
+### python
+
+As of now we support two environments for python `local` and `docker`.
+
+* `local` is a python that is located in your current virtualenv
+* `docker` is python that is located inside your docker image of service(`python_service`).
```
Comment (ron8mcr) [inline README.md:195]:
```suggestion
* `docker` is python that is located inside your docker image of service (`python_service`).
```

File: README.md [hunk 4]
```diff
@@ -186,3 +186,123 @@ Add hosts to `/etc/hosts`.
...
+PYTHON_ENV=docker inv python.run-python --command="--version"
+```
+
+#### run_python
+
+Run python command depending on `PYTHON_ENV` variable(`docker` or `local`).
+
+Settings:
+* `python_entry` python entry command (Default: `python`)
+* `python_service` python service name (Default: `web`)
```
Comment (ron8mcr) [inline README.md:211]:
```suggestion
* `python_docker_service` python service name (Default: `web`)
```
What do you think about this renaming? I think this name is more explicit

File: README.md [hunk 5]
```diff
@@ -186,3 +186,123 @@ Add hosts to `/etc/hosts`.
...
+Run `manage.py` with specified command.
+
+This command also handle starting of required services and waiting DB to
+be ready.
+
+Requires [django_probes](https://github.com/painless-software/django-probes#basic-usage)
+
+#### makemigrations
+
+Run `makemigrations` command and chown created migrations(only for docker env).
```
Comment (ron8mcr) [inline README.md:227]:
```suggestion
Run `makemigrations` command and chown created migrations (only for docker env).
```

File: README.md [hunk 6]
```diff
@@ -186,3 +186,123 @@ Add hosts to `/etc/hosts`.
...
+
+#### check_new_migrations
+
+Check if there is new migrations or not.
```
Comment (ron8mcr) [inline README.md:231]: Add note that result should be checked by exit code

File: README.md [hunk 7]
```diff
@@ -186,3 +186,123 @@ Add hosts to `/etc/hosts`.
...
+
+Settings:
+* `celery_local_cmd` command for celery (Default: `celery --app config.celery:app worker --beat --scheduler=django --loglevel=info`)
+* `celery_service_name` name of celery service (Default: `celery`)
+
+### open-api
+
+#### validate-swagger
+
+Check that generated open_api spec is valid.
```
Comment (ron8mcr) [inline README.md:307]: Add note that this command uses https://github.com/tfranzel/drf-spectacular and it's default validator

File: README.md [hunk 8]
```diff
@@ -186,3 +186,123 @@ Add hosts to `/etc/hosts`.
...
+#### shell
+
+Shortcut for manage.py shell command.
+
+Settings:
+* `shell_command` command to start python shell (Default: `shell_plus --ipython`)
+
+#### dbshell
+
+Open database shell with credentials from either local or dev env.
```
Comment (ron8mcr) [inline README.md:276]:
```suggestion
Open database shell with credentials from current django settings.
```
As I do not see any mentions of settings in parameters of this command

File: saritasa_invocations/django.py [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+    printing.print_success("Django: Make migrations")
+    manage(context, command="makemigrations")
+    if python.get_python_env() == python.PythonEnv.DOCKER:
+        system.chown(context)
+
+
+@invoke.task
+def check_new_migrations(context: invoke.Context) -> None:
+    """Check if there is new migrations or not."""
+    printing.print_success("Checking migrations")
```
Comment (ron8mcr) [inline saritasa_invocations/django.py:68]:
```suggestion
    printing.print_success("Django: Checking migrations")
```

File: saritasa_invocations/django.py [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+
+@invoke.task
+def createsuperuser(
+    context: invoke.Context,
+    email: str = "",
+    username: str = "",
+    password: str = "",
+) -> None:
+    """Create superuser."""
+    printing.print_success("Create superuser")
```
Comment (ron8mcr) [inline saritasa_invocations/django.py:120]:
```suggestion
    printing.print_success("Django: Create superuser")
```

File: saritasa_invocations/python.py [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+    """All possible python environments."""
+
+    DOCKER = "docker"
+    LOCAL = "local"
+
+
+def get_python_env() -> PythonEnv:
+    """Get python environment."""
+    python_env = os.environ.get("PYTHON_ENV", PythonEnv.LOCAL.value).lower()
+    supported_env = tuple(PythonEnv)
```
Comment (ron8mcr) [inline saritasa_invocations/python.py:20]:
```suggestion
    supported_envs = tuple(PythonEnv)
```

**Review by ron8mcr — APPROVED** [review]

File: README.md [hunk 12]
```diff
@@ -186,3 +186,124 @@ Add hosts to `/etc/hosts`.
...
+Shortcut for manage.py shell command.
+
+Settings:
+* `shell_command` command to start python shell (Default: `shell_plus --ipython`)
+
+#### dbshell
+
+Open database shell with credentials from current django settings.
+
+### FastAPI
```
Comment (denismaitak) [inline README.md:278]: FastAPI is capitalized but django, celery, python aren't...

**Review by denismaitak — APPROVED** [review]

**Review by NikAzanov — APPROVED** [review]
As for me, this readme is not very searchable

Let's say I want to find the settings for `inv django.run`, but I can't find anything on `django.run`. And there are many different `run` commands for other modules

But I cannot suggest how to improve it other than

```markdown
### Django

#### django.run

description

#### django.migrate

description
```

But in this case, there is a lot of repetition

**Pull request merged**